app/controllers/admins_controller.rb

```diff
@@ -0,0 +1,55 @@
+class AdminsController < ApplicationController
+
+rescue from ActiveRecord::RecordNotFound, with: :admin_not_found
+
+  #getting all the admins
+  def index
+    admins = Admin.all
+    render json: admins, status: :ok
+  end
+
+  #posting a new admin
+  def create
+    admin = Admin.create(admin_params)
+    render json: admin, status: :created
+  end
+
+  #patching a new admin
+  def update
+    admin = find_admin
+    admin.update(admin_params)
+    render json: admin, status: :accepted
+  end
+
+  #gatting an admin
+  def show
+    admin = find_admin
+    render json: admin, status: :ok
+  end
+
+  #deleting an admin
+  def destroy
+    admin = find_admin
+    admin.destroy
+    head :no_content
+  end
+
+  #the private methods
+  private
+
+  #find the admin
+  def find_admin
+    Admin.find(params(:id))
+  end
+
+  allowed params
+  def admin_params
+    params.permit(:admin_name)
```

Comment:
Same to users, you missed adding all the admin params. The admins params are username, password, password_confirmation. It should be this way `params.permit(:username, :password, :password_confirmation)`
Kndly make these few changes